internal/k8s/watch.go

```diff
@@ -2,23 +2,59 @@ package k8s
 
 import (
 	"context"
+	"fmt"
 
+	"github.com/pkg/errors"
 	"github.com/windmilleng/tilt/internal/model"
 
 	"k8s.io/api/core/v1"
+	apiErrors "k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/watch"
 )
 
-func (kCli K8sClient) WatchPods(ctx context.Context, ls labels.Selector) (<-chan *v1.Pod, error) {
-	ch := make(chan *v1.Pod)
+type watcherFactory func(namespace string) watcher
+type watcher interface {
+	Watch(options metav1.ListOptions) (watch.Interface, error)
+}
 
+func (kCli K8sClient) makeWatcher(f watcherFactory, ls labels.Selector) (watch.Interface, error) {
 	// passing "" gets us all namespaces
-	watcher, err := kCli.core.Pods("").Watch(metav1.ListOptions{LabelSelector: ls.String()})
-	if err != nil {
+	watcher, err := f("").Watch(metav1.ListOptions{LabelSelector: ls.String()})
+	if err == nil {
+		return watcher, nil
+	}
+
+	// If the request failed, we might be able to recover.
+	statusErr, isStatusErr := err.(*apiErrors.StatusError)
+	if !isStatusErr {
 		return nil, err
 	}
 
+	status := statusErr.ErrStatus
+	if status.Code == 403 {
```

Comment:
This is fine as-is, but I wonder if it'd be worth getting the namespace from the k8s entities instead of from the kube context. The current solution leaves us with a bit of a corner case:
* user's kube context is set to `default`
* user deploys service to namespace `foo`
* user lacks access to watch namespace `bar`
* watching `""` fails because of `bar`, so tilt falls back to watching `default`, resulting in tilt never getting pod status and never reporting an error, so the user is kinda left with no information other than "tilt doesn't know what's going on"

I think this PR is still an improvement over status quo, though.